### PR TITLE
EcsServerClusterProvider refactored and tested

### PR DESCRIPTION
**Pull request by dkirillov**
@BrunoCarrier

### CODE REVIEW

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java [hunk 1]
```diff
@@ -523,6 +397,15 @@ public EcsServerCluster getCluster(String application, String account, String na
     return null;
   }
 
+  /**
+   * Temporary implementation to satisfy the interface's implementation.
+   * This will be modified and updated properly once we finish the POC
+   */
+  @Override
+  public EcsServerCluster getCluster(String application, String account, String name, boolean includeDetails) {
```
Comment (BrunoCarrier) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java]: Mmmm the javadoc confuses me

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java [hunk 2]
```diff
@@ -523,6 +397,15 @@ public EcsServerCluster getCluster(String application, String account, String na
...
+    return getCluster(application, account, name);
+  }
+
   /**
    * Temporary implementation to satisfy the interface's implementation.
    * This will be modified and updated properly once we finish the POC
```
Comment (BrunoCarrier) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java]: Confusing javadoc

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java (hunk 1 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java]: Done.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java (hunk 2 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java]: Done.

**Comment by BrunoCarrier** [comment]
LGTM +2

**Pull request merged**